crates/dojo-erc/src/erc20/erc20.cairo

```diff
@@ -175,6 +187,35 @@ mod ERC20 {
             self.emit(Transfer { from: Zeroable::zero(), to: recipient, value: amount });
         }
 
+        fn _increase_allowance(
+            ref self: ContractState, spender: ContractAddress, added_value: u256
+        ) -> bool {
+            let caller = get_caller_address();
+            let token = get_contract_address();
+            assert(!spender.is_zero(), 'ERC20: approve to 0');
+            assert(!caller.is_zero(), 'ERC20: approve from 0');
```

Comment:
I think these should go in the system, and then you have mostly no need for a 'private implementation' function since everything is delegated to the system